apps/notifications/src/alert/services/deployment-balance-alerts/deployment-balance-alerts.service.ts

```diff
@@ -0,0 +1,108 @@
+import { Injectable } from '@nestjs/common';
+import template from 'lodash/template';
+
+import { ChainBlockCreatedDto } from '@src/alert/dto/chain-block-created.dto';
+import {
+  DeploymentBalanceAlertOutput,
+  DeploymentBalanceAlertRepository,
+} from '@src/alert/repositories/deployment-balance-alert/deployment-balance-alert.repository';
+import { ConditionsMatcherService } from '@src/alert/services/conditions-matcher/conditions-matcher.service';
+import { DeploymentService } from '@src/alert/services/deployment/deployment.service';
+import { BrokerService } from '@src/broker';
+import { LoggerService } from '@src/common/services/logger.service';
+
+type AlertCallback = (
+  alert: DeploymentBalanceAlertOutput,
+) => Promise<void> | void;
+
+@Injectable()
+export class DeploymentBalanceAlertsService {
+  constructor(
+    private readonly alertRepository: DeploymentBalanceAlertRepository,
+    private readonly conditionsMatcher: ConditionsMatcherService,
+    private readonly brokerService: BrokerService,
+    private readonly deploymentService: DeploymentService,
+    private readonly loggerService: LoggerService,
+  ) {
+    this.loggerService.setContext(DeploymentBalanceAlertsService.name);
+  }
+
+  private readonly DEPLOYMENT_BALANCE_BLOCKS_THROTTLE = 10;
```

Comment:
Ah ok interesting, so we're checking the balance only every 10 blocks or so (every minute). I think this is a fair delay to be notified for the balance alert, but yea if we wanted to do it in real time it would have to be a little different like we discussed.

Reply:
I'm not sure I follow, how different should it be? This is a var that can be changed to match our needs